src/v/storage/segment_appender.h

```diff
@@ -109,6 +109,7 @@ class segment_appender {
     size_t _bytes_flush_pending{0};
     ss::semaphore _concurrent_flushes;
     ss::lw_shared_ptr<chunk> _head;
+    ss::lw_shared_ptr<ss::semaphore> _prev_head_write;
```

Comment:
I think then we should have a `struct head_page{}` to wrap this into a cohesive unit.